### ISSUE TITLE
MarshalToString on nil panics instead of throwing error.

### ISSUE THREAD

**Issue by ElizabethU** [golang/protobuf#462]
If I'm working with an object with optional fields, and I have a line of code

`string, err := (&jsonpb.Marshaler{OrigName: true}).MarshalToString(object.GetOptionalObject())`

and the optional object isn't there, calling `MarshalToString` on `nil` doesn't collect the error in the `err` value where I can handle it, it panics with the message:

`panic: reflect: call of reflect.Value.Field on zero Value`

I see there's already have handle a similar issue on [call of reflect.Value.Interface on zero Value](https://github.com/golang/protobuf/issues/33) and think the solution would be similar.

**Comment by puellanivis** [comment]
You should be checking to ensure that GetOptionalObject returns non-nil rather than just passing it blindly into MarshalToString.

It is a known and documented return value that GetOptionalObject can return nil, which means that the object doesn’t appear.

The correct behavior of MarshalToString((*ObjectType)(nil)) would also still be expected to be "{}" not returning an error value.

**Comment by dsnet** [comment]
\cc @cybrcodr to decide whether to do anything here.

**Comment by ElizabethU** [comment]
Yup, I am checking it, it just seems that as we're using this all over our code base, somewhere, someone will forget to check for nil, and I'd prefer it not panic when that happens.

Returning `{}` sounds like what should be expected, but raises another interesting question, as I'm actually getting the nil from the autogenerated code from that's the get method for a field that's a protobuf. When nothing is provided, it returns nil rather than an empty protobuf, which would seem more precise. It's possible there's lots of good reasons for that which I just haven't seen, and in any case it's out of the scope of this bug.

I'd love to see this return `{}` so I can handle that instead of a panic.

**Comment by cybrcodr** [comment]
I agree that it should not panic.  It should return an error instead similar to `proto.Marshal`.

I don't think it should return `{}` as that would pertain to an empty object, which is different from a nil.

**Comment by cybrcodr** [comment]
Just sent out PR #469.
